.github/workflows/nightly-benchmark.yml

```diff
@@ -21,30 +21,58 @@ jobs:
   benchmark:
     runs-on:
       group: fuelcore-benchmark
+    concurrency:
+      group: fuelcore-benchmark # only allow one benchmark run at a time
+      cancel-in-progress: false
     outputs:
       benchmark_results: ${{ steps.benchmarks.outputs.benchmark_results }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+
       - name: Install Rust Toolchain
         uses: dtolnay/rust-toolchain@master
         with:
           toolchain: ${{ env.RUST_VERSION }}
           targets: "wasm32-unknown-unknown"
+
+      - name: Install criterion
+        run: cargo install cargo-criterion
+
       - name: Cache
         uses: Swatinem/rust-cache@v2
+
       - name: Run benchmarks
-        id: benchmarks
+        id: run_benchmarks
         run: |
           cargo criterion -p fuel-core-benches --message-format json --bench vm > nightly_benchmarks.json
           cargo run -p fuel-core-benches --bin collect --release -- --input nightly_benchmarks.json --format consensus-parameters --output nightly_benchmarks.json
-          echo "benchmark_results=$(cat nightly_benchmarks.json | tr -d '\n' | jq '.${{ env.CONSENSUS_PARAMETERS_VERSION }}.gas_costs')" >> "$GITHUB_OUTPUT" # this will fail when consensus parameters version changes
+
+      - name: Archive benchmark results
+        uses: actions/upload-artifact@v4
+        with:
+          name: nightly_benchmarks.json
+          path: nightly_benchmarks.json
+
+      - name: Set benchmark result output
+        id: benchmarks
+        run: | # this will fail when consensus parameters version changes
+          echo "benchmark_results=$(cat nightly_benchmarks.json | tr -d '\n' | jq '.${{ env.CONSENSUS_PARAMETERS_VERSION }}.gas_costs')" >> "$GITHUB_OUTPUT"
+
   create_pr:
     needs: benchmark
     runs-on: ubuntu-latest
-    if: ${{ github.event.inputs.create_pr == 'true' }}
+    permissions:
+      pull-requests: write
+
+    if: ${{ github.event.inputs.create_pr == 'true' || github.event_name == 'pull_request' }}
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install GitHub CLI
```

Comment:
Just realized this isn't needed since it's running on a GitHub runner, not a self-hosted one.

Reply:
addressed in [b71985d](https://github.com/FuelLabs/fuel-core/pull/2922/commits/b71985de3e6858c1509b2df1892017093fa33559)